code/modules/client/verbs/viewport.dm

```diff
@@ -0,0 +1,48 @@
+/// Used in OnResize() - Holds the clients prefs.
+GLOBAL_VAR_INIT(lock_client_view_x, null)
+GLOBAL_VAR_INIT(lock_client_view_y, null)
+GLOBAL_LIST_INIT(valid_icon_sizes, list(32, 48, 64, 96, 128))
+/// General handler for viewport updates. Takes an argument for an icon size, if not aligning with your preference, updates your preference.
+/client/verb/SetWindowIconSize(val as num|text)
+	set hidden = 1
+	winset(src, "mapwindow.map", "icon-size=[val]")
+	if(prefs && val != prefs.read_preference(/datum/preference/numeric/icon_size))
+		prefs.write_preference(GLOB.preference_entries[/datum/preference/numeric/icon_size], val)
+		prefs.save_preferences()
+	OnResize()
+/// Verb called by the games skin.dmf to update the server to update view.
```

Comment:
this is kinda a personal thing but I'd appreciate spaces between "blocks" of code, like functions
Makes this sort of thing easier to read, not standardized just me complaining